application/aam-backend-service/src/main/kotlin/com/aamdigital/aambackendservice/reporting/reportcalculation/controller/ReportCalculationController.kt

```diff
@@ -244,7 +257,9 @@ class ReportCalculationController(
             ).byteInputStream()
 
             while ((bodyStream.read(buffer).also { bytesRead = it }) != -1) {
-                outputStream.write(buffer, 0, bytesRead)
+                if (bytesRead > 0) {
```

Comment:
out of curiosity: why is this condition needed?

Reply:
just make sure to write when they actually exist. should not be a problem in most cases, just double check